tungsten_ci_utils/generate_build_change_info/last_successful.py

```diff
@@ -0,0 +1,95 @@
+import sys
+import argparse
+import MySQLdb
+import logging
+import json
+
+log = logging.getLogger(__name__)
+
+def set_logging():
+    log.setLevel(logging.DEBUG)
+    formatter = logging.Formatter(
+        '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    console = logging.StreamHandler()
+    console.setLevel(logging.DEBUG)
+    console.setFormatter(formatter)
+    log.addHandler(console)
+
+set_logging()
+
+def get_json_data(file):
+    json_file = open(file).read()
+    data = json.loads(json_file)
+    return data
+
+def close_db_exit_err(db):
+    if db is not None:
+        db.close()
+    sys.exit(1)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--credentials-json", action="append")
+    parser.add_argument("branch")
+    parser.add_argument("build_number")
+    args = parser.parse_args()
+    branch = str(args.branch)
+    credentials_json = args.credentials_json[0]
+
+    build_number = og_build_number = int(args.build_number)
```

Comment:
Set "int" type of the "build_number" argument to get automatic validation and conversion.